### PR TITLE
python310Packages.edlib: init at 1.3.9

### PR DESCRIPTION
**Pull request by bcdarwin**
edlib: init at unstable-2021-08-20

(Note that the C++ code and Python bindings are maintained in the same repo but with separate version numbers; hence the C++ code getting an `unstable-` version here corresponding to the commit introducing version `1.3.9` in the Python bindings even though the same `src` is used for both.)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [NA] (Module updates) Added a release notes entry if the change is significant
  - [NA] (Module addition) Added a release notes entry if adding a new NixOS module
  - [NA] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/science/biology/edlib/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib, stdenv, fetchFromGitHub, cmake }:
+
+stdenv.mkDerivation rec {
+  pname = "edlib";
+  version = "unstable-2021-08-20";
```
Comment (jbedo) [inline pkgs/development/libraries/science/biology/edlib/default.nix:5]: Why not use the latest release?

File: pkgs/development/libraries/science/biology/edlib/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    runHook preCheck
+    bin/runTests
+    runHook postCheck
+  '';
+
+  meta = with lib; {
+    homepage = "https://martinsos.github.io/edlib";
+    description = "Lightweight, fast C/C++ library for sequence alignment using edit distance";
+    maintainers = with maintainers; [ bcdarwin ];
+    license = licenses.mit;
```
Comment (jbedo) [inline pkgs/development/libraries/science/biology/edlib/default.nix:27]: `platforms` is missing

File: pkgs/development/python-modules/edlib/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+, pythonOlder
+, fetchFromGitHub
+, edlib
+, cython
+, python
+}:
+
+buildPythonPackage {
+  inherit (edlib) pname src meta;
+  version = "1.3.9";
```
Comment (jbedo) [inline pkgs/development/python-modules/edlib/default.nix:12]: If using the latest release instead of unstable then this version can be inherited too.

**Review by jbedo — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/edlib/default.nix (hunk 3 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/edlib/default.nix:12]: The issue is the python and c++ version numbers don't match.

File: pkgs/development/libraries/science/biology/edlib/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/libraries/science/biology/edlib/default.nix:27]: Added, thanks.

File: pkgs/development/libraries/science/biology/edlib/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/libraries/science/biology/edlib/default.nix:5]: See below.

**Review by jbedo — APPROVED** [review]

File: pkgs/development/python-modules/edlib/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  disabled = pythonOlder "3.6";
+
+  sourceRoot = "source/bindings/python";
+
+  preBuild = ''
+    cp -r ${edlib.src}/edlib .
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/edlib/default.nix:19]: If we have the source already, we not cd or symlink?

File: pkgs/development/python-modules/edlib/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  EDLIB_OMIT_README_RST = 1;
+  EDLIB_USE_CYTHON = 1;
+
+  nativeBuildInputs = [ cython ];
+  buildInputs = [ edlib ];
+
+  checkPhase = ''
+    runHook preCheck
+    ${python.interpreter} test.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/edlib/default.nix:30]: pytest is not supported.

Comment (bcdarwin) [reply pkgs/development/python-modules/edlib/default.nix:30]: Indeed, it is not.

File: pkgs/development/python-modules/edlib/default.nix (hunk 4 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/edlib/default.nix]: Changed to `ln -s` instead of `cp`, thanks.

**Pull request merged**